src/platform/packages/private/kbn-controls-renderer/src/components/floating_actions.tsx

```diff
@@ -104,7 +107,7 @@ export const FloatingActions: FC<FloatingActionsProps> = ({
         const newActions: FloatingActionItem[] = currentActions?.filter(
           (current) => current.id !== action.id
         );
-        if (isCompatible) {
+        if (isCompatible && (disabledActions ?? []).indexOf(action.id) === -1) {
```

Comment:
This ensures that the disabled actions are filtered out even when the actions change via the `compatibilityChangesSubject`